### PR TITLE
AB#79260: dashboard filter new options

### PR DESCRIPTION
**Pull request by guisoares1**
# Description

Made necessary changes to the queries so that the new filter saving logic (moving from application level to dashboard level) works correctly.

Additionally, a migration was carried out so that old records were corrected.

## Useful links

- [Link to ticket](https://dev.azure.com/WHOHQ/EMSSAFE/_workitems/edit/79260)
- [Link to front-end branch](https://github.com/ReliefApplications/ems-frontend/pull/2078)

## Type of change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)
- [X] Improvement (refactor or addition to existing functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.

- [X] I ran the migrations and observed the system behavior.

# Checklist:

( * == Mandatory ) 

- [x] * I have set myself as assignee of the pull request
- [x] * My code follows the style guidelines of this project
- [x] * Linting does not generate new warnings
- [x] * I have performed a self-review of my own code
- [x] * I have put the ticket for review, adding the oort-backend team to the list of reviewers
- [x] * I have commented my code, particularly in hard-to-understand areas
- [x] * I have put JSDoc comment in all required places
- [x] * My changes generate no new warnings
- [x] * I have included screenshots describing my changes if relevant
- [x] * I have selected labels in the Pull Request, according to the changes with code brings
- [ ] I have made corresponding changes to the documentation ( if required )
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

# More explanation
https://www.loom.com/share/05a716d61b9744faaf51fb304c21d1e5?sid=f87cf896-582a-4f76-93ae-8ceed801b145

### CODE REVIEW

File: src/models/dashboard.model.ts [hunk 1]
```diff
@@ -22,6 +22,10 @@ export interface Dashboard extends Document {
   archived: boolean;
   archivedAt?: Date;
   gridOptions?: any;
+  filterVariant?: string;
```
Comment (AntoineRelief) [inline src/models/dashboard.model.ts:25]: put all that under a filter property

File: src/models/dashboard.model.ts [hunk 2]
```diff
@@ -22,6 +22,10 @@ export interface Dashboard extends Document {
...
+  closable?: boolean;
+  contextualFilter?: any;
```
Comment (AntoineRelief) [inline src/models/dashboard.model.ts:27]: no need to call that contextualFilter now if we have a filter property
we can name it like "structure"
and contextualFilterPosition can be just "position"

File: src/models/dashboard.model.ts [hunk 3]
```diff
@@ -42,6 +46,8 @@ const dashboardSchema = new Schema<Dashboard>(
     name: String,
     structure: mongoose.Schema.Types.Mixed,
     showFilter: Boolean,
```
Comment (AntoineRelief) [inline src/models/dashboard.model.ts:48]: showFilter should be "show" and put under the new filter property

File: src/schema/mutation/editDashboard.mutation.ts [hunk 4]
```diff
@@ -74,14 +82,22 @@ export default {
...
         args.structure && { structure: args.structure },
         args.name && { name: args.name },
         !isNil(args.showFilter) && { showFilter: args.showFilter },
         args.buttons && { buttons: args.buttons },
-        args.gridOptions && { gridOptions: args.gridOptions }
+        args.gridOptions && { gridOptions: args.gridOptions },
+        args.filterVariant && { filterVariant: args.filterVariant },
+        !isNil(args.closable) && { closable: args.closable },
+        args.contextualFilter && { contextualFilter: args.contextualFilter },
+        args.contextualFilterPosition && {
```
Comment (AntoineRelief) [inline src/schema/mutation/editDashboard.mutation.ts:98]: see comments above

File: src/models/dashboard.model.ts (hunk 2 above)
Comment (estelafs) [reply src/models/dashboard.model.ts:27]: We already have a property called "structure" in the dashboard model (used for widgets), I will call the contextualFilter of "filterStructure"

File: migrations/1700653450166-set-contextualfilter-to-dashboard-level.ts [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+ * @returns just migrate data.
+ */
+export const up = async () => {
+  await startDatabaseForMigration();
+  const applications = await Application.find()
+    .populate({
+      path: 'pages',
+      model: 'Page',
+    })
+    .select('name pages');
```
Comment (AntoineRelief) [inline migrations/1700653450166-set-contextualfilter-to-dashboard-level.ts:47]: @estelafs @guisoares1 
I'll do it during the review, but be aware of select in mongoose
there, you don't get the contextualFilter from the application, as it's not part of the list
so you're sure that the dashboard filter will be empty for all dashboards

**Pull request merged**